packages/mobile/src/app/selectors.ts

```diff
@@ -108,6 +113,89 @@ export const linkBankAccountEnabledSelector = (state: RootState) => state.app.li
 
 export const sentryTracesSampleRateSelector = (state: RootState) => state.app.sentryTracesSampleRate
 
+export const supportedBiometryTypeSelector = (state: RootState) => state.app.supportedBiometryType
+
+export const biometryEnabledSelector = (state: RootState) =>
+  state.app.biometryEnabled && !!state.app.supportedBiometryType
+
+export const useBiometrySelector = (state: RootState) => state.app.useBiometry
+
+export const activeScreenSelector = (state: RootState) => state.app.activeScreen
+
 export const dappsListApiUrlSelector = (state: RootState) => state.app.dappListApiUrl
 
 export const superchargeButtonTypeSelector = (state: RootState) => state.app.superchargeButtonType
+
+type StoreWipeRecoveryScreens = Extract<
+  Screens,
+  | Screens.NameAndPicture
+  | Screens.ImportWallet
+  | Screens.VerificationEducationScreen
+  | Screens.VerificationInputScreen
+>
+type CreateAccountScreens = Extract<
+  Screens,
+  | Screens.NameAndPicture
+  | Screens.PincodeSet
+  | Screens.VerificationEducationScreen
+  | Screens.VerificationInputScreen
+>
+type RestoreAccountScreens = CreateAccountScreens & Screens.ImportWallet
+
+export const storeWipeRecoverySteps: { [key in StoreWipeRecoveryScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.ImportWallet]: 2,
+  [Screens.VerificationEducationScreen]: 3,
+  [Screens.VerificationInputScreen]: 3,
+}
+export const createAccountSteps: { [key in CreateAccountScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.PincodeSet]: 2,
+  [Screens.VerificationEducationScreen]: 3,
+  [Screens.VerificationInputScreen]: 3,
+}
+export const restoreAccountSteps: { [key in RestoreAccountScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.PincodeSet]: 2,
+  [Screens.ImportWallet]: 3,
+  [Screens.VerificationEducationScreen]: 4,
+  [Screens.VerificationInputScreen]: 4,
+}
+
+// The logic in this selector should be moved to a hook when all registration
+// screens are function components
+export const registrationStepsSelector = createSelector(
```

Comment:
This selector is a bit tricky with all the different cases!

I'm not sure what the best way to implement this is, but here's a suggestion, feel free to ignore it. It's more boilerplate-y and less generic but I feel like easier to read and understand.

The idea is to just re-write the code 3 times for each of the 3 flows. I don't remember exactly but I think this is what the original version looked like? In that case I'm sorry for making your life difficult :( So something like:

```
if (recoveringFromStoreWipe) {
  return {
    step: storeWipeRecoverySteps[activeScreen]
    totalSteps: 3,
  }
} else if (chooseRestoreAccount) {
  const totalSteps = biometryEnabled ? 5 : 4
  let step = restoreAccountSteps[activeScreen] // Note that I'd add the BiometricScreen to the restoreAccountSteps object with the number 4.
  if (biometricsEnabled && isVerificationScreen(activeScreen)) {
    step = 5
  }
  return { totalSteps, step }
} else {
  const totalSteps = biometryEnabled ? 4 : 3
  let step = createAccountSteps[activeScreen] // Note that I'd add the BiometricScreen to the createAccountSteps object with the number 3.
  if (biometricsEnabled && isVerificationScreen(activeScreen)) {
    step = 4
  }
  return { totalSteps, step }
}
```

I don't think this is the best option either to be honest, not sure how to do it in a way that's tidy and scalable into the future, but I think this is good for now? To me particularly having the separate `if` for each flow helps me read that particular block thinking just about that one flow which is easier than juggling the three flows in a more generic fashion, but maybe it's just me.

In the existing code, the part that made me think too hard and I don't fully understand is the `if (biometryEnabled)`, since we're handling two flows there I think the indexes might be slightly wrong, but maybe I'm wrong.


In any case, if the current code works feel free to leave as it is as I'm not convinced that my version is too much better either :/

Reply:
i agree! in the original i had the blocks separated, i can go back to that. i do think it's clearer like what you have above with each flow separated, especially after adding the recover from store wipe flow. separate blocks makes it looks less magic that the numbers add up in the end (i did write some unit tests for the flows, so the current version does work but readability is key so i will update!)

Reply:
hopefully the update is more clear? i took your suggestion to add the biometrics screen to the objects. i think let's not spend too much time on it because it sounds like there is work in the pipeline to remove the verification steps from the onboarding altogether, so this logic would become redundant in the near future :)